### PR TITLE
add line of code for bundle audit CVEs

### PR DESCRIPTION
**Pull request by ghbren**
Add line of code for bundle/npm/yarn audit CVEs.

Tested with
* new unit specs
* docker run local salus and manually examined txt, json and sarif outputs

### CODE REVIEW

File: lib/salus/scanners/bundle_audit.rb [hunk 1]
```diff
@@ -44,6 +45,18 @@ def run_scanner(scanner, ignore)
         hash = serialize_vuln(result)
         @vulns.push(hash)
 
+        pattern = case hash[:type]
+                  when 'UnpatchedGem'
+                    hash[:name] + ' (' + hash[:version] + ')'
+                  when 'InsecureSource'
+                    hash[:source]
+                  end
+        if !pattern.nil?
```
Comment (joshua-ostrom) [inline lib/salus/scanners/bundle_audit.rb:54]: Minor, can you move the line_number logic to a stand alone method

File: lib/salus/scanners/yarn_audit.rb [hunk 2]
```diff
@@ -46,6 +46,7 @@ def run
         # vulns were all whitelisted
         return report_success if vulns.empty?
 
+        add_line_number(vulns)
```
Comment (joshua-ostrom) [inline lib/salus/scanners/yarn_audit.rb:49]: Another minor but I'd be a fan of moving the logic you've added here to a stand alone class for working with the line number pieces.  We will have a coupling of passing the vulns hash around but if we document the required hash keys I think it'll be a nice separation of logic here.
```
YarnLock.new('yarn.lock').add_line_number(vulns)
```

File: lib/salus/scanners/bundle_audit.rb (hunk 1 above)
Comment (ghbren) [reply lib/salus/scanners/bundle_audit.rb:54]: done

File: lib/salus/scanners/yarn_audit.rb (hunk 2 above)
Comment (ghbren) [reply lib/salus/scanners/yarn_audit.rb]: done

File: lib/salus/package_utils.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Dir.entries(File.expand_path('package_utils', __dir__)).sort.each do |filename|
```
Comment (joshua-ostrom) [inline lib/salus/package_utils.rb:1]: :+1 for the sort... that race condition still haunts me :-)

File: lib/salus/package_utils/gemfile_lock.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    def add_line_number(cve)
+      pattern = case cve[:type]
+                when 'UnpatchedGem'
+                  cve[:name] + ' (' + cve[:version] + ')'
+                when 'InsecureSource'
+                  cve[:source]
+                end
+      if !pattern.nil?
+        line_in_gemfile_lock = `grep -n " #{pattern}" #{@path}`
```
Comment (joshua-ostrom) [inline lib/salus/package_utils/gemfile_lock.rb:15]: `IO.popen(["grep", "-n", pattern, @path]) ` 
would offer some protection against any malicious content... that's a small risk here but I think if someone hijacked the vuln database and injected a malicious string as the "version" we could pick it up here.   Probably paranoid to worry about it but I'm becoming a fan of IO.popen over backticks. 

Thoughts?  If I'm being too paranoid ignore me!

**Comment by joshua-ostrom** [comment]
One small security callout, see what you think... I think your test coverage will suffice since we refactored here it's more of an end-end test of the new class than unit coverage but it looks like the behavior is tested.  

Thanks Brianna!

**Review by joshua-ostrom — APPROVED** [review]

**Pull request merged**